File: 08-2-2-Adapter/Ducks/MallardDuck.cs
namespace Adapter.Ducks
{
    using System;
    using Adapter.Fly;
    using Adapter.Quack;

    public sealed class MallardDuck : Duck
    {
        public MallardDuck()
            : base(new DuckQuackStrategy(), new RealFlyStrategy())
        {
        }

        public override void Display() => Console.WriteLine("Green head, gray-brown body.");
    }
}

File: 09-DesignPatterns-Java/Solution/Adapter/src/Ducks/MallardDuck.java
package Ducks;

import Fly.RealFlyStrategy;
import Quack.DuckQuackStrategy;

public class MallardDuck extends Duck {
    public MallardDuck() {
        super(new DuckQuackStrategy(), new RealFlyStrategy());
    }

    @Override
    public void display() {
        System.out.println("Green head, gray-brown body.");
    }
}
